### ISSUE TITLE
Using an invalid type in an "on" clause results in multiple errors, some confusing

### ISSUE THREAD

**Issue by Hixie** [dart-lang/sdk#54235]
I forgot to import `dart:io` and wrote this code:

```dart
    ...
  } on FormatException {
    ...
  } on SocketException { // errors below are on this line
    ...
```

The error messages from the analyzer included:

* `A potentially nullable type can't be used in an 'on' clause because it isn't valid to throw a nullable expression`
* `Dead code: This on-catch block won't be executed because 'InvalidType' is a subtype of 'FormatException' and hence will have been caught already`
* `The name 'SocketException' isn't a type and can't be used in an on-catch clause`

Only one of these makes sense, the other two are very confusing and misleading.

**Comment by lrhn** [comment]
I'm assuming the third one is the one that makes sense.

**Comment by bwilkerson** [comment]
Yes. We obviously have some work to do to suppress extraneous diagnostics when an `InvalidType` is encountered.
